app/api/groups/add_child.robustness.feature

```diff
@@ -0,0 +1,140 @@
+    When I send a POST request to "/groups/13/add_child/11"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Child group is UserAdmin
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/13/add_child/14"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Child group is Root
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/13/add_child/15"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Child group is RootSelf
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/13/add_child/16"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Child group is RootAdmin
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/13/add_child/17"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Parent group is UserSelf
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/18/add_child/11"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Relation already exists
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/13/add_child/11"
+    Then the response code should be 400
+    And the response error message should contain "The relation already exists"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: A group cannot become an ancestor of itself
+    Given I am the user with ID "4"
+    When I send a POST request to "/groups/11/add_child/13"
+    Then the response code should be 403
+    And the response error message should contain "A group cannot become an ancestor of itself"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
```

Comment:
missing test: 

- [ ] given parent and child are the same. I know it should fail on ancestors for the moment but if the ancestor cache stuff changes, this should still be prevented, so at least a test should validate it.
- [ ] the "removed" sType case (maybe in `add_child.feature`)

Reply:
The first one actually returns "insufficient access rights" error

Reply:
added a scenario for the first case

Reply:
added a scenario for relation rewriting